tianshou/data/utils/converter.py

```diff
@@ -94,6 +94,12 @@ def to_torch_as(
 Hdf5ConvertibleType = Dict[str, Hdf5ConvertibleValues]  # type: ignore
 
 
+def to_hdf5_via_pickle(x: object, y: h5py.Group, key: str) -> None:
+    """Pickle, convert byte stream to numpy array and write to HDF5 dataset."""
+    data = np.frombuffer(pickle.dumps(x), dtype=np.byte)
```

Comment:
I would rather make it a nested method, since it is never intended to be called manually, and it is very short.

Reply:
You are right it belongs into `to_hdf5`. I have turned it into an inner function.